src/main/java/de/embl/cba/mobie/ui/viewer/SourcesPanel.java

```diff
@@ -706,13 +711,14 @@ public void removeSourceFromPanelAndViewers( String sourceName )
         removeSourceFromPanelAndViewers( sourceNameToSourceAndMetadata.get( sourceName ) );
     }
 
-    public void removeSourceFromPanelAndViewers( SourceAndMetadata< ? > sam )
+    private void removeSourceFromPanelAndViewers( SourceAndMetadata< ? > sam )
     {
         updateSegments3dView( sam.metadata(), this, false );
         updateSource3dView( sam, this, false );
 
         removeSourceFromPanel( sam.metadata().displayName );
 		removeLabelViews( sam.metadata().displayName );
+		sourceNameToSourceAndMetadata.remove( sam.metadata().displayName );
```

Comment:
Seeing this now, maybe we should rename for sake of clarity:
`sourceNameToSourceAndCurrentMetadata`
to make clear that this is something temporary that can be removed?

Other question:
If the user comes back and adds the source a second time to the viewer, which display settings would be applied?
I guess the default ones? Is that what we want? We can leave it like this for now for simplicity.

Reply:
Yes - for the moment, this will apply the default settings, so nothing persists. I guess later, if there are certain things we'd like to persist e.g. contrast settings etc..., then we can make these modify the default metadata directly? The issue with using the current metadata is that everything would persist > additional tables / colouring by columns etc which I'm not sure is what we want..